### PR TITLE
ci: test cargo on `aarch64-gnu`

### PR DESCRIPTION
**Pull request by cuviper**
Since `aarch64-unknown-linux-gnu` is a tier-1 target, we should also test cargo on it, especially since cargo's own CI doesn't cover this yet. This might have helped us discover #123733 sooner, which is not a cargo problem but was uncovered by a new cargo test (which we'll have to skip for now). Everything else passes in my local run, so at least we'll have a guard against future regressions.

### CODE REVIEW

**Comment by cuviper** [comment]
Dropped cargotest, as suggested on [zulip](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/CI.20for.20.60aarch64-unknown-linux-gnu.60/near/432795901).

File: src/ci/docker/host-aarch64/aarch64-gnu/Dockerfile [hunk 1]
```diff
@@ -26,4 +26,6 @@ ENV RUST_CONFIGURE_ARGS \
  --enable-sanitizers \
  --enable-profiler \
  --enable-compiler-docs
-ENV SCRIPT python3 ../x.py --stage 2 test
+# FIXME: Skipping cargo panic_abort_doc_tests due to https://github.com/rust-lang/rust/issues/123733
+ENV SCRIPT python3 ../x.py --stage 2 test && \
+  python3 ../x.py --stage 2 test src/tools/cargo --test-args "--skip panic_abort_doc_tests"
```
Comment (ehuss) [inline src/ci/docker/host-aarch64/aarch64-gnu/Dockerfile:31]:
```suggestion
  python3 ../x.py --stage 2 test src/tools/cargo --test-args \"--skip panic_abort_doc_tests\"
```

Comment (cuviper) [reply src/ci/docker/host-aarch64/aarch64-gnu/Dockerfile:31]: Thanks! I could have sworn I tested this locally, but maybe I only ran it without the skip to re-confirm the issue, and forgot to run again...

**Pull request merged**